### PR TITLE
Initial commit for barrier free: node order and css adjustments

### PR DESCRIPTION
**Pull request by edellucien**

### CODE REVIEW

File: src/components/BarrierFree/BarrierFree.js [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+function BarrierFree() {
+  const zoom = useSelector(state => state.map.zoom);
+  const center = useSelector(state => state.map.center);
+  const resolution = useSelector(state => state.map.resolution);
+
+  const map = useSelector(state => state.app.map);
+  const layerService = useSelector(state => state.app.layerService);
+
+  // Filtering function based on generalization levels.
+  const wktFormat = new WKT();
```
Comment (oterral) [inline src/components/BarrierFree/BarrierFree.js]: initalize the format outside the `function BarrierFree`

File: src/components/BarrierFree/BarrierFree.js [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+      .reverse()
+      .filter(l => !l.getIsBaseLayer() && l.getVisible());
+
+    const mapExtent = map.getView().calculateExtent(map.getSize());
+
+    let visFeats = [];
+    visibleLayers.forEach(l => {
+      if (l && l.olLayer) {
+        const features = l.olLayer.getSource().getFeatures();
+        if (features.length > 0) {
```
Comment (oterral) [inline src/components/BarrierFree/BarrierFree.js]:
```suggestion
        if (features.length) {
```

File: src/components/BarrierFree/BarrierFree.js [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+    visibleLayers = layerService
+      .getLayersAsFlatArray()
+      .reverse()
+      .filter(l => !l.getIsBaseLayer() && l.getVisible());
+
+    const mapExtent = map.getView().calculateExtent(map.getSize());
+
+    let visFeats = [];
+    visibleLayers.forEach(l => {
+      if (l && l.olLayer) {
```
Comment (oterral) [inline src/components/BarrierFree/BarrierFree.js]: I don't think this `if` is necessary

File: src/components/BarrierFree/BarrierFree.js [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+        const features = l.olLayer.getSource().getFeatures();
```
Comment (oterral) [inline src/components/BarrierFree/BarrierFree.js]: Use getSource().getFeaturesInExtent

File: src/components/BarrierFree/BarrierFree.js [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+        if (features.length > 0) {
+          features.forEach(f => f.set('layerRef', l));
+          const visibleFeats = features.filter(
+            feat =>
+              containsExtent(mapExtent, feat.getGeometry().getExtent()) &&
```
Comment (oterral) [inline src/components/BarrierFree/BarrierFree.js]: Use getFeaturesInExtent directly from the source. 
https://openlayers.org/en/master/apidoc/module-ol_source_Vector-VectorSource.html#getFeaturesInExtent

File: src/components/BarrierFree/BarrierFree.js [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+    });
+    visibleFeatures = visFeats;
+  };
+
+  const focusFeature = () => {
+    const featToFocus = visibleFeatures[featureIndex];
+    featLayer = featToFocus ? featToFocus.get('layerRef') : null;
+    if (featLayer) {
+      tabFeature = true;
+      featLayer.clickedFeature = featToFocus;
```
Comment (oterral) [inline src/components/BarrierFree/BarrierFree.js]: Where is used the `clickedFeature` ?

File: src/components/BarrierFree/BarrierFree.js [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+
+  useEffect(() => {
+    setVisibleFeats();
+  }, [center, zoom, resolution, layerService, setVisibleFeats]);
+
+  useEffect(() => {
+    // ComponentDidMount
+    document.addEventListener('click', onDocumentClick);
+    document.addEventListener('keyup', onDocumentKeyUp);
+    document.addEventListener('keydown', onDocumentKeyDown);
```
Comment (oterral) [inline src/components/BarrierFree/BarrierFree.js]: I think the functions  onDocumentXXX  must be defined in the `useEffect`. Have you verify if the event is properly remove ?

File: src/components/Dialog/Dialog.scss [hunk 8]
```diff
@@ -7,7 +7,7 @@
     .tm-dialog-container {
       min-width: 320px;
       width: auto;
-      overflow: hidden;
+      overflow: visible;
```
Comment (oterral) [inline src/components/Dialog/Dialog.scss]: Are you sure it hasn't side effects?

File: src/components/TopicsMenuHeader/TopicsMenuHeader.js [hunk 9]
```diff
@@ -60,7 +60,10 @@ const TopicsMenuHeader = ({ isOpen, onToggle }) => {
 
   return (
     <Button
-      className={`wkp-menu-header${isOpen ? ' open' : ''}`}
+      className="wkp-menu-header"
+      style={{
+        borderBottom: isOpen ? '1px solid #eee' : '',
+      }}
```
Comment (oterral) [inline src/components/TopicsMenuHeader/TopicsMenuHeader.js]: Why using a style ?

File: src/components/TrafimageMaps/TrafimageMaps.js [hunk 10]
```diff
@@ -226,17 +222,31 @@ function TrafimageMaps({
...
         validExtent={[656409.5, 5740863.4, 1200512.3, 6077033.16]}
       />
     ),
   };
 
-  const appElements = getComponents(defaultElements, elements);
+  // Declared above the map for tabIndex accessibility.
+  const appTopElements = Object.entries(topElements).map(([k, v]) =>
+    elements[k] ? <div key={k}>{v}</div> : null,
+  );
```
Comment (oterral) [inline src/components/TrafimageMaps/TrafimageMaps.js]: use `getComponents` method.

File: src/components/TrafimageMaps/TrafimageMaps.js [hunk 11]
```diff
@@ -226,17 +222,31 @@ function TrafimageMaps({
...
+
+  const appDefaultElements = Object.entries(defaultElements).map(([k, v]) =>
+    elements[k] ? <div key={k}>{v}</div> : null,
+  );
```
Comment (oterral) [inline src/components/TrafimageMaps/TrafimageMaps.js]: use getComponents method.

File: src/config/layers.js [hunk 12]
```diff
@@ -69,6 +69,7 @@ export const netzkarteLayer = new MapboxLayer({
   radioGroup: 'baseLayer',
   preserveDrawingBuffer: true,
   zIndex: -1, // Add zIndex as the MapboxLayer would block tiled layers (buslines)
+  tabIndex: -1, // Canvas not accessible via Tab navigation.
```
Comment (oterral) [inline src/config/layers.js]: Define it by default in `react-spatial`

File: src/layers/HandicapLayer/HandicapLayer.js [hunk 13]
```diff
@@ -4,6 +4,7 @@ import OLVectorSource from 'ol/source/Vector';
 import GeoJSON from 'ol/format/GeoJSON';
 import WKT from 'ol/format/WKT';
 import { Style, RegularShape, Stroke, Fill } from 'ol/style';
+import getFeatureGeometry from '../../utils/getFeatureGeometry';
```
Comment (oterral) [inline src/layers/HandicapLayer/HandicapLayer.js]: Since it's only useable for `HandicapLayer` maybe call the file `HandicapUtils`

**Comment by mhaertwig** [comment]
@edellucien I did not see that you created this mr in the meantime closed the old one. Sorry!
Old: https://github.com/geops/trafimage-maps/pull/151/files

Anyways, please stop working on this one. I ask the customer if it's really necessary to tab through features on the map. I think accessing features by using the search is much more useful.

@oterral @friedjoff fyi.

**Pull request closed without merging**